tensorflow/core/kernels/scatter_functor.h

```diff
@@ -217,6 +258,30 @@ struct ScatterFunctorBase {
     }
     return -1;
   }
+
+  Index operator()(OpKernelContext* c, const Device& d,
+                   typename TTypes<T>::Matrix params,
+                   typename TTypes<T>::ConstMatrix updates,
+                   typename TTypes<Index>::ConstFlat indices) {
+    // indices and params sizes were validated in DoCompute().
+    const Index N = static_cast<Index>(indices.size());
+    const Index limit = static_cast<Index>(params.dimension(0));
+    const Index min_n_threshold = 1024;
+    const Index ser_par_ratio = 10000;
+    // For parallelizing the updates, duplicate entries need to be handled
+    // correctly.Multiple updates to the same index has to be serialized.
+    // This can lead to lock contention which may nullify the benefits of
+    // parallelization. Assuming uniform random distribution of the indices, we
+    // come up with a rough heuristic and determine whether the updates execute
+    // serially or parallelly. Also if 'N' is small, overheads of parallel
+    // execution outweigh its benefits and hence we check the value of N.
+    const bool execute_serial =
+        ((N < min_n_threshold) || ((N / limit) > ser_par_ratio)) ? true : false;
```

Comment:
The expression already returns a boolean. We can drop the ternary operator.
```suggestion
        ((N < min_n_threshold) || ((N / limit) > ser_par_ratio));
```